### PR TITLE
feat/Implement Keycloak

### PR DESCRIPTION
**Pull request by arranfw**
Limitations:
- If a request is made on the dashboard as soon as a user logs in for the first time (within ~1s), it will fail because tokens aren't ready yet. The work around is to refresh the page. This is can potentially be resolved by adding an additional loading state that waits for tokens from keycloak
  - Addressed  ✅

Todo:
- Extract Keycloak variables to makefile
  - Addressed ✅

### CODE REVIEW

File: client/src/keycloakClient.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Keycloak from 'keycloak-js';
+
+const authUrl = 'https://keycloak.freshworks.club/auth';
+const authRealm = 'PBPG-dev';
+const authClientId = 'PBGP';
```
Comment (arranfw) [inline client/src/keycloakClient.js:5]: Need to extract these to the makefile

**Review by PeterPavlatos — APPROVED** [review]

File: client/src/index.js [hunk 2]
```diff
@@ -40,23 +44,68 @@ const IEMessage = () => {
...
+        </Box>
+      }
+      authClient={keycloakClient}
+      onTokens={handleTokens}
+    >
+      <AuthContext.Provider value={authState}>
+        <ThemeProvider theme={Theme}>
+          <BrowserRouter>
+            <ToastProvider>
+              <AuthProvider authIssuer={authIssuer} userType={AuthIssuer.User}>
```
Comment (karthikktamilmani-freshworks) [inline client/src/index.js:93]: We can remove this provider as we've Keycloak provider now. But can be done as a part of moving the code

**Review by karthikktamilmani-freshworks — APPROVED** [review]
Looks great!

File: client/src/index.js (hunk 2 above)
Comment (arranfw) [reply client/src/index.js:93]: Good point, I'll make a note of that

**Pull request merged**